src/cargo/ops/cargo_add/mod.rs

```diff
@@ -537,19 +538,67 @@ fn get_latest_dependency(
                     std::task::Poll::Pending => registry.block_until_ready()?,
                 }
             };
-            let latest = possibilities
-                .iter()
-                .max_by_key(|s| {
-                    // Fallback to a pre-release if no official release is available by sorting them as
-                    // less.
+
+            fn find_latest<'a>(
+                possibilities: impl Iterator<Item = &'a Summary>,
+            ) -> Option<&'a Summary> {
+                possibilities.max_by_key(|s| {
+                    // Fallback to a pre-release if no official release is available by sorting them
+                    // as less.
                     let stable = s.version().pre.is_empty();
                     (stable, s.version())
                 })
-                .ok_or_else(|| {
-                    anyhow::format_err!(
-                        "the crate `{dependency}` could not be found in registry index."
-                    )
-                })?;
+            }
+
+            let mut latest = find_latest(possibilities.iter()).ok_or_else(|| {
+                anyhow::format_err!(
+                    "the crate `{dependency}` could not be found in registry index."
+                )
+            })?;
+
+            if config.cli_unstable().msrv_policy {
+                if let Some(rust_version) = manifest
+                    .data
+                    .as_table()
+                    .get("package")
+                    .and_then(|p| p.get("rust-version"))
```

Comment:
We can't rely on `LocalManfest` for this because `package.rust-version` might be inherited from the workspace

The caller has `workspace` that we can use to look up `Package` and call [`Package::rust_version`](https://docs.rs/cargo/latest/cargo/core/package/struct.Package.html#method.rust_version)